### ISSUE TITLE
Notifications / automatations start only if app or control center gets opened

### ISSUE THREAD

**Issue by benzman81** [nfarina/homebridge#1006]
I have an issue that often notifications and automatations of multiple homebridge plugins start only if I open the home app or slide up the control center. It might be since update of iOS to 10.2, but I am not sure. I couldn't find any other information on the net about this though. The log states, that the actions already took place (mostly/all background updates through a webhook). So I guess there is a queue where the update to the charesteristics get stuck.

Does any one else have these issues? Any idea?

**Comment by NorthernMan54** [comment]
This would be a design pattern issue with the plugins, and how events are managed between the device and the HomeKit environment.  In most plugins, status of a device is pulled based on a status request from the home app, which is what you are seeing.  To drive notifications, the device would need to generate the change event into HomeKit.  With my plugins I have been polling the device and sending updates to HomeKit and am not seeing the behaviour you are experiencing with them.  My motion sensors generate notifications regardless of my usage of the home app.

To sum this up, these issues should be raised with the plugin developer and ask them properly support notifications.

**Comment by benzman81** [comment]
@NorthernMan54 I am the developer of most of the plugins having this issue. The Characteristics are updated properly from the webhook. The plugins worked before, but now, somehow it doesnt work anymore. My only obeservation is that I updated to iOS 10.2 and added a new plugin. I'll doublecheck if the new plugin is the problem, but it doesnt do crazy things.

**Comment by benzman81** [comment]
Even with the new plugin uninstalled the issue persists. The log tells me, that the code passed the "getCharacteristic(Characteristic.XXX).setValue(newState)" line, but as I said, sometimes the notifications only come through after I open the app.

**Comment by erfinb** [comment]
@Northernman54 what plugin are you referring to that you are using that doesn't experience this issue, and could you take a look at the issue that we are experiencing and see what you think? 🙂

**Comment by NorthernMan54** [comment]
I think the first step is determining of its an issue inside of home kit or homebridge.  If you have the debug log turned from homebridge, but have nothing open on your phone you should see something similar to this for each accessory that is notification capable, when starting up.

  HAPServer [CC:22:3D:E3:CD:35] HAP Request: PUT /characteristics +1ms
  Accessory [Bart] Processing characteristic set: [{"aid":5,"iid":9,"ev":true},{"aid":5,"iid":10,"ev":true}] +0ms
  Accessory [Bart] Registering Characteristic "Current Door State" for events +0ms
  Accessory [Bart] Registering Characteristic "Target Door State" for events +0ms
  EventedHTTPServer [::ffff:192.168.1.208] HTTP Response is finished +0ms

This is the home app on the phone registering itself for changes to these values.

[12/26/2016, 11:49:52 AM] [IOT] GarageDoor is at Fault NODE-869815
  EventedHTTPServer [::ffff:192.168.1.106] Sending HTTP event '5.9' with data: {"characteristics":[{"aid":5,"iid":9,"value":4}]} +6m
  EventedHTTPServer [::ffff:192.168.1.212] Sending HTTP event '5.9' with data: {"characteristics":[{"aid":5,"iid":9,"value":4}]} +3ms
  EventedHTTPServer [::ffff:192.168.1.208] Sending HTTP event '5.9' with data: {"characteristics":[{"aid":5,"iid":9,"value":4}]} +0ms

The first line is my plugin running setValue, and the others are homebridge sending the data to the phone.   When this happened my phone generated a notification of the garage door event.

**Comment by erfinb** [comment]
How would I go about enabling debugging logging on? 🙂

**Comment by NorthernMan54** [comment]
There are a couple of ways, take a look at this https://github.com/nfarina/homebridge/wiki/Basic-Trouble-Shooting

For me the easiest is to start from the command line with this option

DEBUG=* homebridge -D

**Comment by adekka** [comment]
Can it  have to do something with this? [ https://discussions.apple.com/thread/7669326](url)

**Comment by benzman81** [comment]
@adekka no, this discussion is about a lock being a security device. I have this issue with non security devices like contact sensors and occpancy sensors. Still seems to me like a bug in homekit introduced with ios 10.2.

**Comment by gijoecool** [comment]
I have the same problem with z-wave door contacts and homebridge-openhab. I only installed the contacts after 10.2, so I cannot say if it's bug introduced in 10.2.

**Comment by gijoecool** [comment]
Today I updated my raspberry pi to node version 6.9.2 from 4.0.0 and now I get notifications from contact sensors! I think I may have been using an older version of homebridge as well.

**Comment by adekka** [comment]
Can't wait to test! How did you upgrade?

**Comment by gijoecool** [comment]
@adekka Is your homebridge installation on the latest?

**Comment by adekka** [comment]
@gijoecool yes, i recently updated the version..

**Comment by gijoecool** [comment]
Any luck after the update?

**Comment by adekka** [comment]
nope, didn't do the trick..

**Comment by gijoecool** [comment]
:(

**Comment by adekka** [comment]
i'm waiting voor IOS 8.2.1 with the hope that it's an IOS bug..

**Comment by RXM307** [comment]
You'll be waiting awhile for for 8.2.1 :) hoping you mean 10.2.1 :)

**Comment by adekka** [comment]
😋

**Comment by jay-hankins** [comment]
So the consensus is that it's an iOS 10.2 bug? I have tried different HomeKit apps (like Elgato Eve) which makes me think the issue is more on the side of homebridge. I had scheduled a "Sunset" event in the automations tab of the Home app, and it has only worked once. I can't recall, but I think I was using my phone at the time, but did not have the Home app open. I use my Apple TV 4th generation as a Home Hub, which I assume should pick up on the schedules and send the events to homebridge.

**Comment by andyrooo82** [comment]
I've been playing about and although my automations are working at the moment, time will be the judge of whether this is resolved.

What I've Done:

1. Signed out of iCloud in Homekit Hub and back in, reboot Homekit Hub.

2. Removed my crontab which used to restart my Pi ever 12 hours. I thought that Homebridge might be the issue so wanted it to refresh every 12 hours. I think this actaully makes it worse and Homekit Hub doesn't like it when Homebridge is restarted. Get it's knickers in a right twist.

3. I'm currently thinking along the lines of Homekit  becomes "stale". The house is empty most of the day some homekit isn't being used, then when i get home I'm expecting the lights to turn on (Homekit gives the cold sholder). I've setup HTTP-Webhook, IFTTT and Maker Channel to every 15 mins turn a dummy switch On or Off. e.g. at 00-On, 15-Off, 30-On, 45-Off. This (fingers crossed) at the moment is keeping Homekit Hub "Alive" and runs my automations (so far) without failing.

**Comment by adekka** [comment]
Nice approach! Will try the samen!

**Comment by andyrooo82** [comment]
4 days in and still ok

I forgot to mention that another thing I did was instead of using the scenes Arrive Home and Leave Home scenes in my automations I created ones called On and Off (but using them for the same purpose). 

For some reson and I haven't a clue why when I was troubleshooting I found that the scenes we're not being automated when I was using Arrive/Leave Home scenes.

**Comment by erfinb** [comment]
I'm taking your lead and created new scenes, not based on the original 4 (Good Morning, Good Night, Arrive Home, and Leave Home). I wonder if these four default scenes may potentially have issues with triggers. I will update based on the experiences I have with trigger reliability. :-)

**Comment by andyrooo82** [comment]
Yeah good shout @erfnib. I haven't replaced the Good Morning or Good Night scenes as the night scene for me wasn't automated and I was using a timed automation for good morning. The timed automations seemed to fire ok for me. It was just automations based on sensor triggers e.g arrive home and leave home. 

I haven't had any issues since

**Comment by andyrooo82** [comment]
Yeah good shout @erfnib I haven't replaced the Good Morning or Good Night scenes as the night scene for me wasn't automated and I was using a timed automation for good morning. The timed automations seemed to fire ok for me. It was just automations based on sensor triggers e.g arrive home and leave home. 

I haven't had any issues since

**Comment by erfinb** [comment]
Sadly enough, the automation seems more spotty than ever.  I'm running the newest betas on my Apple TV as well as my iPhone.  I have to open the Home app to get the triggers to work with most of the time.  So, I'm out of ideas on this.  I even tried having a WeMo switch turn on/off every 15 minutes to see if that would help as suggest before with no success.

**Comment by andyrooo82** [comment]
Yeah I have now had some failed automations. Both times it was the leave scene. Arrive Home seems to be ok. I've rewritten my leave scene in Hesperus in a slightly different way to see if this helps. 

This is truly frustrating though and Apple need to sort it out. The whole point of HomeKit to to automate your home. If it cannot fire automations without fail then it's about as much good and nuns tits. 

I might start a chocolate making company and sell vegetables!!

**Comment by Merlinuf** [comment]
Have the same problem error log getting spammed

EventedHTTPServer [::ffff:10.0.2.124] Sending HTTP event '5.9' with data: {"characteristics":[{"aid":5,"iid":9,"value":1}]}

**Comment by 0ldF0x** [comment]
I had same problem. You're probably missing event characteristic:
Somehing like:
this.service.getCharacteristic(Characteristic.MotionDetected).updateValue(true)
(updateValue instead of setValue). 
Works perfect for me.

**Comment by benzman81** [comment]
Changed to updateValue, but still issues with contact sensor and locks. Motion and smoke sensors are working.

**Comment by ooii** [comment]
This is probably due to issues with your homekit hub not working properly.

**Comment by softlion** [comment]
I have the exact same problem. In the Homebridge Nut plugin. When the UPS is disconnected, i don't receive any notification on the iphone unless i open the Home app. So this means the plugin creator did not implement notification correctly ?
